src/nusoap.php

```diff
@@ -8433,7 +8410,7 @@ function __construct($cache_dir='.', $cache_lifetime=0) {
      * @access private
      */
     function createFilename($wsdl) {
-        return $this->cache_dir.'/wsdlcache-' . md5($wsdl);
+        return $this->cache_dir.'/wsdlcache-' . password_hash($wsdl, PASSWORD_DEFAULT);
```

Comment:
This is not a security hash, so `password_hash` is inappropriate - password hashing is designed to be slow, but in this case we want something _fast_. `md5` is still perfectly appropriate for this kind of use.